### DISCUSSION TITLE
I am getting this error in using private key

### DISCUSSION THREAD

**Discussion by heheboii11** [Cyfrin/foundry-full-course-cu#1216] [Q&A]
ramireddykarthikreddy@Ramireddys-MacBook-Air foundry-simple-storage % forge script script/DeploySimpleStorage.s.sol --rpc-url http:/127.0.0.1:8545 --private-key $PRIVATE_KEY --broadcast
error: a value is required for '--private-key <RAW_PRIVATE_KEY>' but none was supplied

For more information, try '--help'.
![Uploading Image 18-12-23 at 3.02 PM.jpeg…]()

**Comment by ghost** [comment]
Did you run `source .env` before running the command? Also check the name of the variable in the `.env`.

**Comment by heheboii11** [comment]
I did run source .env but I am getting this error .env:1: 0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d not found after running source .env

**Comment by xyz899** [comment]
Hi,
Remove 0x from your private key, check if your var in `.env` is correctly called  `$PRIVATE_KEY` and if none works chezck the last video o Cyfrin about never having to use again a .env file, for security purposes, we use cast wallet check here : https://www.youtube.com/watch?v=VQe7cIpaE54&ab_channel=CyfrinAudits

**Comment by heheboii11** [comment]
Thank you using encryption helped  me

**Reply by xyz899** [reply]
write the reply as answer so people know its solved and can get the help directly

**Reply by heheboii11** [reply]
These did not help but I used the encryption method the link above I used the same process

**Reply by xyz899** [reply]
so the reply did help, reply is the full answer
